receiver/splunkhecreceiver/receiver.go

```diff
@@ -231,20 +253,73 @@ func (r *splunkReceiver) Shutdown(context.Context) error {
 	return err
 }
 
-func (r *splunkReceiver) writeSuccessResponse(ctx context.Context, resp http.ResponseWriter, eventCount int) {
+func (r *splunkReceiver) writeSuccessResponseWithAck(ctx context.Context, resp http.ResponseWriter, eventCount int, ackID uint64) {
+	r.writeSuccessResponse(ctx, resp, eventCount, []byte(fmt.Sprintf(responseOKWithAckID, ackID)))
+}
+
+func (r *splunkReceiver) writeSuccessResponseWithoutAck(ctx context.Context, resp http.ResponseWriter, eventCount int) {
+	r.writeSuccessResponse(ctx, resp, eventCount, okRespBody)
+}
+
+func (r *splunkReceiver) writeSuccessAckResponse(ctx context.Context, resp http.ResponseWriter, acks map[uint64]bool) {
+	ackString, _ := json.Marshal(acks)
+	r.writeSuccessResponse(ctx, resp, 0, []byte(fmt.Sprintf(ackResponse, ackString)))
+}
+
+func (r *splunkReceiver) writeSuccessResponse(ctx context.Context, resp http.ResponseWriter, eventCount int, bodyContent []byte) {
 	resp.Header().Set(httpContentTypeHeader, httpJSONTypeHeader)
 	resp.WriteHeader(http.StatusOK)
-	if _, err := resp.Write(okRespBody); err != nil {
+	if _, err := resp.Write(bodyContent); err != nil {
 		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, eventCount, err)
 	}
 }
 
+func (r *splunkReceiver) handleAck(resp http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	if req.Method != http.MethodPost {
+		r.failRequest(ctx, resp, http.StatusBadRequest, invalidMethodRespBodyPostOnly, 0, errInvalidMethod)
+		return
+	}
+
+	// shouldn't run into this case since we only enable this handler IF ackExt exists. But we have this check just in case
+	if r.ackExt == nil {
+		r.failRequest(ctx, resp, http.StatusInternalServerError, errInternalServerError, 0, errExtensionMissing)
+		return
+	}
+
+	// check that the channel exists
+	partitionID := req.Header.Get(splunk.HTTPSplunkChannelHeader)
+	if len(partitionID) == 0 {
+		r.failRequest(ctx, resp, http.StatusBadRequest, requiredDataChannelHeader, 0, nil)
+		return
+	}
+
+	// check validity of channel
+	_, err := uuid.Parse(partitionID)
+	if err != nil {
+		r.failRequest(ctx, resp, http.StatusBadRequest, invalidDataChannelHeader, 0, err)
+		return
+	}
+
+	dec := json.NewDecoder(req.Body)
+	var ackRequest splunk.AckRequest
```

Comment:
i think we are missing a validation that the array for `acks` is not empty. Can you add a unit test after as well?

Reply:
isn't what you're talking about this case? https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/31767/files#diff-b54d599277d2b73d9fe14051ddb024bec9d3d470a710c14b273e5f3635cfe818R1228-R1243

Reply:
Are we ok here?